coordinator/coordinator.go

```diff
@@ -823,54 +792,106 @@ func (c *Coordinator) runFeatureMaterializeJob(resID metadata.ResourceID, schedu
-			ResourceID:    provider.ResourceID{Name: resID.Name, Variant: resID.Variant, Type: provider.Feature},
-			VType:         provider.ValueTypeJSONWrapper{ValueType: vType},
-			Cloud:         runner.LocalMaterializeRunner,
-			IsUpdate:      true,
-		}
-		serializedUpdate, err := scheduleMaterializeRunnerConfig.Serialize()
-		if err != nil {
-			return fmt.Errorf("serialize materialize runner config: %v", err)
-		}
-		jobRunnerUpdate, err := c.Spawner.GetJobRunner(runner.MATERIALIZE, serializedUpdate, resID)
-		if err != nil {
-			return fmt.Errorf("creating materialize job schedule job runner: %v", err)
-		}
-		cronRunner, isCronRunner := jobRunnerUpdate.(kubernetes.CronRunner)
-		if !isCronRunner {
-			return fmt.Errorf("kubernetes runner does not implement schedule")
-		}
-		if err := cronRunner.ScheduleJob(kubernetes.CronSchedule(schedule)); err != nil {
-			return fmt.Errorf("schedule materialize job in kubernetes: %v", err)
-		}
-		if err := c.Metadata.SetStatus(context.Background(), resID, metadata.READY, ""); err != nil {
-			return fmt.Errorf("set succesful update status for materialize job in kubernetes: %v", err)
-		}
+		return fmt.Errorf("could not update status for materialize job: %v", err)
+	}
+	return nil
+}
+
+func (c *Coordinator) materializeFeature(id metadata.ResourceID, config runner.MaterializedRunnerConfig) error {
+	c.Logger.Infow("Starting Feature Materialization", "id", id)
+	serialized, err := config.Serialize()
+	if err != nil {
+		return fmt.Errorf("could not serialize materialization config: %v", err)
+	}
+	jobRunner, err := c.Spawner.GetJobRunner(runner.MATERIALIZE, serialized, id)
+	if err != nil {
+		return fmt.Errorf("could not get job runner for feature %s: %w", id, err)
+	}
+	completionWatcher, err := jobRunner.Run()
+	if err != nil {
+		return fmt.Errorf("failed to run job: %w", err)
+	}
+	if err := completionWatcher.Wait(); err != nil {
+		return fmt.Errorf("failed to complete job: %w", err)
+	}
+	return nil
+}
+
+func (c *Coordinator) materializeFeatureOnSchedule(id metadata.ResourceID, config runner.MaterializedRunnerConfig, schedule string) error {
+	c.Logger.Infow("Scheduling Feature Materialization", "id", id)
+	config.IsUpdate = true
+	serialized, err := config.Serialize()
+	if err != nil {
+		return fmt.Errorf("serialize materialize runner config: %v", err)
+	}
+	jobRunnerUpdate, err := c.Spawner.GetJobRunner(runner.MATERIALIZE, serialized, id)
+	if err != nil {
+		return fmt.Errorf("creating materialize job schedule job runner: %v", err)
+	}
+	cronRunner, isCronRunner := jobRunnerUpdate.(kubernetes.CronRunner)
+	if !isCronRunner {
+		return fmt.Errorf("kubernetes runner does not implement schedule")
+	}
+	if err := cronRunner.ScheduleJob(kubernetes.CronSchedule(schedule)); err != nil {
+		return fmt.Errorf("schedule materialize job in kubernetes: %v", err)
+	}
+	return nil
+}
+
+func (c *Coordinator) materializeFeatureViaS3Import(id metadata.ResourceID, config runner.MaterializedRunnerConfig) error {
```

Comment:
Should have a check in here that makes sure that the source filestore is actually s3. Was considering doing it later but feel like it should get checked before we attempt to spun up a worker